tools/tpm2_createprimary.c

```diff
@@ -91,17 +96,20 @@ static bool set_name_alg(TPMI_ALG_HASH halg, TPM2B_PUBLIC *public) {
 
 static bool set_alg(TPMI_ALG_PUBLIC type, TPM2B_PUBLIC *public) {
 
-
     switch(type) {
     case TPM2_ALG_RSA: {
         TPMS_RSA_PARMS *r = &public->publicArea.parameters.rsaDetail;
-       r->symmetric.algorithm = TPM2_ALG_AES;
-       r->symmetric.keyBits.aes = 128;
-       r->symmetric.mode.aes = TPM2_ALG_CFB;
-       r->scheme.scheme = TPM2_ALG_NULL;
-       r->keyBits = 2048;
-       r->exponent = 0;
-       public->publicArea.unique.rsa.size = 0;
+        if (public->publicArea.objectAttributes & TPMA_OBJECT_SIGN_ENCRYPT) {
```

Comment:
I have no idea what these algorithms are really for, but you can't have RSA + AES with the attributes on sign

Reply:
@williamcroberts I'm also not completely sure what these algorithms are for. By reading the specs it seems that they are used to define the algorithm and scheme used that can be used by restricted keys and the symmetric algorithm used to decrypt a child key when loading to the TPM.

You can find information about it in:

`12.2 Public Area Structures` and `12.2.3.5 TPMS_RSA_PARMS` in `Trusted Platform Module Library Part 2: Structures` and `25 Object Attributes` in `Trusted Platform Module Library Part 1: Architecture`. 

Could you please mention in the commit message what's the goal of this change? What exactly did you try that didn't work, etc.

Reply:
I tried to make a primary object I could use for signing and verifying:
This command wouldn't work: tpm2_createprimary -Q -G rsa -g sha256 -a o -C parent.ctx -A "fixedtpm|fixedparent|sensitivedataorigin|userwithauth|decrypt|sign"

Based on what you're saying above, maybe it's if not restricted.... but the check in the simulator was on sign.

Reply:
@williamcroberts I could misunderstood the spec too. But it would be nice if you can add that information in the commit message (command executed, errors shown, etc) to have a better understanding on why the change is needed.